### PR TITLE
Issue 88543: TextField labelText doesn't get hintTextStyle when labelTextStyle is non-null Fixed

### PR DESCRIPTION
**Pull request by varunkamanibosc**
This PR fixes issue #88543. According to [labelText](https://api.flutter.dev/flutter/material/InputDecoration/labelText.html) docs, it uses the `hintStyle` when the label is on top of the text field.

Added new condition to check if label widget is null then the label should use the `hintStyle` when the label is on top of the text field. Fixed a test that expects to use `labelStyle` for the given scenario.

The following test cases are already checking the above cases (`packages/flutter/test/material/input_decorator_test.dart`):

1. InputDecorationTheme style overrides
2. InputDecorationTheme labelStyle overrides label widget styles when the widget is a text widget

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by varunkamanibosc** [comment]
Hi @HansMuller Can you please review the PR. Thanks

**Review by HansMuller — COMMENTED** [review]
This looks OK to me however it may disturb existing (internal Google) golden image tests. 

@justinmc - could you verify that this change is safe?

File: packages/flutter/test/material/input_decorator_test.dart [hunk 1]
```diff
@@ -5104,4 +5104,37 @@ void main() {
     // Verify that the styles were passed along
     expect(getLabelStyle(tester).color, labelStyle.color);
   });
+
+  testWidgets('InputDecoration hintStyle is used for labelText when labelText is on top of the input field', (WidgetTester tester) async {
```
Comment (justinmc) [inline packages/flutter/test/material/input_decorator_test.dart:5108]: Is this testing the same case for the label style as the previous test?

What if you add another `pumpWidget` at the end of this test with `isFocused` set to false and check that the label's style is now `labelStyle`?

**Review by justinmc — COMMENTED** [review]
Thanks for fixing this!  Check my comment on your test below.

I'll run the Google tests on this to see if there are any failures.  If so, I'll just update any golden images, shouldn't be a problem.

File: packages/flutter/test/material/input_decorator_test.dart (hunk 1 above)
Comment (varunkamanibosc) [reply packages/flutter/test/material/input_decorator_test.dart:5108]: Okay I'll do that

Comment (varunkamanibosc) [reply packages/flutter/test/material/input_decorator_test.dart:5108]: @justinmc This test case is already checking for the case when `isFocused` is set to false.

Comment (varunkamanibosc) [reply packages/flutter/test/material/input_decorator_test.dart:5108]: I'm assuming you mistyped that and you want to pump another widget with `isFocused` set to true and that label should have `labelStyle` applied.

Comment (justinmc) [reply packages/flutter/test/material/input_decorator_test.dart]: Yes, sorry for the confusion!

**Review by justinmc — APPROVED** [review]
LGTM 👍 

There was a failure that looked unrelated, rerunning.

**Comment by varunkamanibosc** [comment]
Thanks @justinmc and @HansMuller 
What is the next step?

**Comment by justinmc** [comment]
Running the Google tests, then the flutter-gold tests should run.  If they both go green then we can merge.

**Comment by justinmc** [comment]
This did cause a few Google tests to fail, but the changes were almost unnoticeable.  I think it was a slight change in color of the hint maybe.  I'll update the test images after this is merged.

**Comment by justinmc** [comment]
The gold tests are stuck at pending, but I'll go ahead and merge this.

**Pull request merged**